### ISSUE TITLE
"mode": 'jsx-only' forbids any string literal inside interpolation blocks

### ISSUE THREAD

**Issue by lgenzelis** [edvardchen/eslint-plugin-i18next#109]
I'm using `"eslint-plugin-i18next": "6.0.1"`.

My use case: I want to forbid string literals in children and aria-labels. So, this is my config:
```javascript
'i18next/no-literal-string': [
  'error',
  {
    "mode": 'jsx-only',
    'jsx-attributes': { include: ['aria-label'] },
  },
],
```

This works fine, except that it detects **any** string literal inside an interpolation block as an error. For example:

```javascript
const list = ['a', 'b', 'c'];

function Foo() {
  console.log('rending foo'); // ----> This is fine
  return (
    <div>
      {list.map((item) => {
        console.log('debugging', item); // -------> An error is reported in the word 'debugging'
        return <div key={item}>{item}</div>;
      })}
    </div>
  );
}
```

**Comment by edvardchen** [comment]
`jsx-only` mode did check everything surrounded by JSX markup. 

My initial thought is that people might make mistakes easily like 
```jsx
<>hello world</>  // error
// vs
<>{"hello world"}</> // but this's fine?
```

**Comment by lgenzelis** [comment]
Ok, I get it. In your example, both should be errors, indeed. The difference, I guess, is that in my example the "wrongly" detected error is inside a callback (for `.map()`). I don't know any details about the implementation, but maybe you could check if a callback is being declared, and in that scenario, "forget" about any JSX ancestor?

As a side question, is it ok that I need to specify `"mode": 'jsx-only',` for my use case? Or is that a bug? I mean, if I just do 
```
'i18next/no-literal-string': [
  'error',
  {
    'jsx-attributes': { include: ['aria-label'] },
  },
],
```
then `aria-label`s are not checked at all.

**Comment by edvardchen** [comment]
The default mode is `jsx-text-only` which can be found [here](https://github.com/edvardchen/eslint-plugin-i18next/blob/main/lib/options/defaults.json). 

> but maybe you could check if a callback is being declared, and in that scenario, "forget" about any JSX ancestor?

Actually the behavior of mode `jsx-text-only` is very close to this. But It only checks literals JSX text like `<>hello world</>`

Since the existing mode `all` `jsx-only` `jsx-text-only` couldn't satisfy your need, I think it needs a new mode which checks only literal JSX text and JSX attributes

**Comment by lgenzelis** [comment]
I gave it a try @edvardchen ! 😁 Please let me know what you think https://github.com/edvardchen/eslint-plugin-i18next/pull/110 🙏

**Comment by edvardchen** [comment]
It will be a breaking change since we change the behavior of mode `jsx-only`

Your proposal would make function calls `<>{display("hello")} </>` `<>[0].map(()=>"hello")</>` **valid** but `<>{["name"][0]}</>` **invalid** which is a bit confusing. Hard to understand why. It would feel like we had left something unfinished. 

Instead, I prefer to check or not to check these dynamic part at all. We can make a new mode named `jsx-static-only` which checks only JSX text and attributes

**Comment by saravanan-inc** [comment]
@lgenzelis, @edvardchen, is there any updates on this, or any solution to it?

**Comment by lgenzelis** [comment]
I'm just ignoring the rule (with an `//  eslint-disable-next-line` ) when I run into this issue. 😅
